### PR TITLE
[BEAM-90] TestCountingSource can throw on checkpointing

### PR DESCRIPTION
**Pull request by mshields822**

### CODE REVIEW

**Comment by mshields822** [comment]
R=klk
Urgent: blocking customer impacting bugfix.

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/dataflow/TestCountingSource.java [hunk 1]
```diff
@@ -46,31 +48,41 @@
...
 public class TestCountingSource
     extends UnboundedSource<KV<Integer, Integer>, TestCountingSource.CounterMark> {
+  private static final Logger LOG = LoggerFactory.getLogger(TestCountingSource.class);
+
   private static List<Integer> finalizeTracker;
   private final int numMessagesPerShard;
   private final int shardNumber;
   private final boolean dedup;
+  private boolean throwOnFirstSnapshot;
+  private static boolean thrown = false;
```
Comment (kennknowles) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/dataflow/TestCountingSource.java]: Can you leave a little comment here to explain?

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/dataflow/TestCountingSource.java [hunk 2]
```diff
@@ -46,31 +48,41 @@
  */
 public class TestCountingSource
     extends UnboundedSource<KV<Integer, Integer>, TestCountingSource.CounterMark> {
+  private static final Logger LOG = LoggerFactory.getLogger(TestCountingSource.class);
+
   private static List<Integer> finalizeTracker;
   private final int numMessagesPerShard;
   private final int shardNumber;
   private final boolean dedup;
+  private boolean throwOnFirstSnapshot;
```
Comment (kennknowles) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/dataflow/TestCountingSource.java]: This can be final.

**Comment by mshields822** [comment]
PTAL

**Pull request closed without merging**